HW17/package0/ThreadSafeList.java

```diff
@@ -0,0 +1,45 @@
+package package0;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class ThreadSafeList<T> {
```

Comment:
The class must implement the List interface in the thread save manner. The class must implement the List interface in a thread-safe manner. Methods add, get and remove must be implemented. The rest of the methods can be empty/not implemented (for example, throw OperationNotSupportedException).